### PR TITLE
Samaritan 2129/lcv last month 1 4

### PR DESCRIPTION
**Pull request by dannysperry**
[Trello Card](https://trello.com/c/0tIf8Z9J/2129-lcv-last-month-1-4)

**Description**
Adding Completable Goals and Needs to `<LastCheckin />`

### CODE REVIEW

File: src/components/individuals/autocomplete.js [hunk 1]
```diff
@@ -6,8 +6,6 @@ import { withStyles } from '@material-ui/core/styles';
 import TextField from '@material-ui/core/TextField';
 import Paper from '@material-ui/core/Paper';
 import MenuItem from '@material-ui/core/MenuItem';
-import { connect } from 'react-redux';
-import { compose } from 'redux';
```
Comment (dannysperry) [inline src/components/individuals/autocomplete.js]: This is a cleanup file. Autocomplete had a redux state connector that wasn't supposed to be there anymore. We are passing the props from the Individual component which connects to redux through the Individual container.

File: src/components/individuals/index.js [hunk 2]
```diff
@@ -283,24 +284,6 @@ Individuals.propTypes = {
     homelesses:       PropTypes.object,
     homelessInfo:     PropTypes.object,
     homelessTimeline: PropTypes.object,
-    homelessGoals: PropTypes.oneOfType([
```
Comment (dannysperry) [inline src/components/individuals/index.js]: These were throwing warnings, which means I didn't have them setup right. We have a card for fixing PropTypes throughout, for now I'm just removing the requirements till we work on that effort

File: src/components/individuals/index.js [hunk 3]
```diff
@@ -167,15 +154,26 @@ class Individuals extends Component {
...
         } else if (homelessInfo && homelessTimeline) {
             return (
-                <IndividualProfile {...this.props} openDrawer={this.openDrawer} />
+                <IndividualProfile
+                    homelessInfo={homelessInfo}
+                    homelessGoals={homelessGoals}
+                    homelessNeeds={homelessNeeds}
+                    isFetchingHomelessGoals={isFetchingHomelessGoals}
+                    isFetchingHomelessNeeds={isFetchingHomelessNeeds}
+                    openDrawer={this.openDrawer} />
```
Comment (dannysperry) [inline src/components/individuals/index.js]: passing state through props

File: src/components/individuals/index.js [hunk 4]
```diff
@@ -199,13 +197,16 @@ class Individuals extends Component {
...
+
+        if (!isFetchingHomelessInfo && isObjectEmpty(homelessInfo)) {
             return (<div></div>)
-        } else if (this.props.isFetchingHomelessInfo) {
+        } else if (isFetchingHomelessInfo) {
             return (<CircularProgress/>)
-        } else if (this.props.homelessInfo) {
-            return (<LifecareVisit holder={this.props.homelessInfo} closeDrawer={this.closeDrawer} />)
+        } else if (homelessInfo) {
+            return (<LifecareVisit holder={homelessInfo} closeDrawer={this.closeDrawer} />)
```
Comment (dannysperry) [inline src/components/individuals/index.js]: Cleaned up variables/this.props usage

File: src/components/individuals/profile/index.js [hunk 5]
```diff
@@ -36,7 +38,12 @@ class IndividualProfile extends Component {
 
                 <Redeem holder={homelessInfo} />
                 <Lifecare openDrawer={this.props.openDrawer} holder={homelessInfo} />
-                <Status holder={homelessInfo} goals={homelessGoals} needs={homelessNeeds} />
+                <Status
+                    holder={homelessInfo}
+                    goals={homelessGoals}
+                    needs={homelessNeeds}
+                    isFetchingHomelessGoals={isFetchingHomelessGoals}
+                    isFetchingHomelessNeeds={isFetchingHomelessNeeds} />
```
Comment (dannysperry) [inline src/components/individuals/profile/index.js]: passing state through props

File: src/components/individuals/profile/index.js [hunk 6]
```diff
@@ -48,17 +55,6 @@ class IndividualProfile extends Component {
...
-    homelessGoals: PropTypes.oneOfType([
-      PropTypes.bool,
-      PropTypes.object,
-    ]),
-    homelessNeeds: PropTypes.oneOfType([
-      PropTypes.bool,
-      PropTypes.object,
-    ]),
-    isFetchingHomelessGoals: PropTypes.bool.isRequired,
-    isFetchingHomelessNeeds: PropTypes.bool.isRequired,
```
Comment (dannysperry) [inline src/components/individuals/profile/index.js]: These were throwing warnings, which means I didn't have them setup right. We have a card for fixing PropTypes throughout, for now I'm just removing the requirements till we work on that effort

File: src/components/individuals/profile/status/index.js [hunk 7]
```diff
@@ -114,17 +114,6 @@ class Status extends Component {
...
-      PropTypes.bool,
-      PropTypes.object,
-    ]),
-    needs: PropTypes.oneOfType([
-      PropTypes.bool,
-      PropTypes.object,
-    ]),
-    holder: PropTypes.object.isRequired,
-    isFetchingHomelessGoals: PropTypes.bool.isRequired,
-    isFetchingHomelessNeeds: PropTypes.bool.isRequired,
```
Comment (dannysperry) [inline src/components/individuals/profile/status/index.js]: These were throwing warnings, which means I didn't have them setup right. We have a card for fixing PropTypes throughout, for now I'm just removing the requirements till we work on that effort

File: src/components/individuals/autocomplete.js (hunk 1 above)
Comment (cvkumar) [reply src/components/individuals/autocomplete.js]: Ahh yes thanks for this!

File: src/components/lifecare_visit/screens/form/last_checkin/goals/index.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    render() {
+        const {
+            classes,
+            holder,
+            goals,
+            isFetchingHomelessGoals,
+            completeGoalAction
+        } = this.props
+
+        let goalsComponents = null
```
Comment (cvkumar) [inline src/components/lifecare_visit/screens/form/last_checkin/goals/index.js]: Would putting below logic in `componentDidMountI()` method be better?

File: src/redux/actions/index.js [hunk 9]
```diff
@@ -312,6 +313,43 @@ function setHomelessGoals(response) {
...
+      .then(response => {
+          if (response.ok) {
+              return response.json()
+          }
+          throw new Error(`${response.status}: ${response.statusText}`)
+      })
+      .then(responseJson => {
+          dispatch(removeHomelessGoal(responseJson.goal_action.id))
+      })
+      .catch(error => {
```
Comment (cvkumar) [inline src/redux/actions/index.js]: Nit: May want to add new state to handle this in the future.

File: src/util/index.js [hunk 10]
```diff
@@ -23,3 +32,46 @@ export const getMonth = (i) => {
     }
     return months[i]
 }
+
+var ones = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
+var tens = ['', '', 'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety'];
```
Comment (cvkumar) [inline src/util/index.js]: missing `ten`?

File: src/util/index.js [hunk 11]
```diff
@@ -23,3 +32,46 @@ export const getMonth = (i) => {
     }
     return months[i]
 }
+
+var ones = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
```
Comment (cvkumar) [inline src/util/index.js]: I remember solving this in college x)

**Pull request merged**